mlflow/pytorch/__init__.py

```diff
@@ -176,51 +252,89 @@ def log_model(
+
             def forward(self, x):
                 y_pred = self.linear(x)
-            return y_pred
-        # our model
-        model = Model()
-        criterion = torch.nn.MSELoss(size_average=False)
-        optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+                return y_pred
+
+        def gen_data():
+
+            # Example linear model modified to use y = 2x
+            # from https://github.com/hunkim/PyTorchZeroToAll
+            # X training data, y labels
+            X = torch.arange(1.0, 25.0).view(-1, 1)
+            y = torch.from_numpy(np.array([x * 2 for x in X])).view(-1, 1)
+            return X, y
+
+        # Define model, loss, and optimizer
+        model = LinearNNModel()
+        criterion = torch.nn.MSELoss()
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.001)
+
         # Training loop
-        for epoch in range(500):
-            # Forward pass: Compute predicted y by passing x to the model
-            y_pred = model(x_data)
-            # Compute and print loss
-            loss = criterion(y_pred, y_data)
-            print(epoch, loss.data.item())
-            #Zero gradients, perform a backward pass, and update the weights.
+        epochs = 250
+        X, y = gen_data()
+        for epoch in range(epochs):
+            # Forward pass: Compute predicted y by passing X to the model
+            y_pred = model(X)
+
+            # Compute the loss
+            loss = criterion(y_pred, y)
+
+            # Zero gradients, perform a backward pass, and update the weights.
             optimizer.zero_grad()
             loss.backward()
             optimizer.step()
 
-        # After training
-        for hv in [4.0, 5.0, 6.0]:
-            hour_var = torch.Tensor([[hv]])
-            y_pred = model(hour_var)
-            print("predict (after training)",  hv, model(hour_var).data[0][0])
-        # log the model
+        if (epoch + 1) % 50 == 0:
+            print('Epoch: {}/{}, loss: {:.4f}'.format(epoch + 1., epochs, loss.data.item()))
+
+        # Log the model
         with mlflow.start_run() as run:
-            mlflow.log_param("epochs", 500)
-            mlflow.pytorch.log_model(model, "models")
+            mlflow.pytorch.log_model(model, "models_pth")
 
-            # logging scripted module
+            # convert to scripted model and log the model
             scripted_pytorch_model = torch.jit.script(model)
-            mlflow.pytorch.log_model(scripted_pytorch_model, "models")
+            mlflow.pytorch.log_model(scripted_pytorch_model, "scripted_models_pth")
+
+        # fetch the logged model artifacts
+        print("--")
+        print("run_id: {}".format(run.info.run_id))
+        for artifact_path in ["models_pth/data", "scripted_models_pth/data"]:
+            artifacts = [f.path for f in MlflowClient().list_artifacts(run.info.run_id,
+                        artifact_path)]
+            print("artifacts: {}".format(artifacts))
```

Comment:
Maybe just showing these lines should be enough?

Reply:
We could use `...` too.

```python
class MyModel(str):
    ...


model = MyModel()

# train the model
...

with mlflow.start_run():
    mlflow.pytorch.log_model(model, "model")
```

Reply:
Thanks for the input. For all but `log_model()` and `autolog_log()`,  I have elided the code. However, for the other two methods, I think, it's good to have a full example that they can try.